### PR TITLE
[5.5] [PSR-16] Implement SimpleCache

### PR DESCRIPTION
**Pull request by deleugpn**
Make Laravel 5.5 compliant with PSR-16

### CODE REVIEW

File: src/Illuminate/Contracts/Cache/Repository.php [hunk 1]
```diff
@@ -3,8 +3,9 @@
 namespace Illuminate\Contracts\Cache;
 
 use Closure;
+use Psr\SimpleCache\CacheInterface;
 
-interface Repository
+interface Repository extends CacheInterface
```
Comment (lucasmichot) [inline src/Illuminate/Contracts/Cache/Repository.php]: I feel that this should be `\Illuminate\Cache\Repository` that implements `Psr\SimpleCache\CacheInterface`

File: src/Illuminate/Cache/Repository.php [hunk 2]
```diff
@@ -115,6 +115,26 @@ public function many(array $keys)
...
     /**
+     * {@inheritdoc}
+     */
+    public function getMultiple($keys, $default = null)
+    {
+        if (is_null($default)) {
+            return $this->many($keys);
+        }
+
+        $default = collect($default);
```
Comment (lucasmichot) [inline src/Illuminate/Cache/Repository.php]: Why not just using an array ?

File: src/Illuminate/Contracts/Cache/Repository.php (hunk 1 above)
Comment (deleugpn) [inline src/Illuminate/Contracts/Cache/Repository.php]: I'm doing here the same thing I did in psr-11:51bfc610692c3ed3388dec0ec04bd3baR8

Comment (lucasmichot) [inline src/Illuminate/Contracts/Cache/Repository.php]: Yeah true, it' weird. I had a look at `\Illuminate\Contracts\Container\Container` too which extends `\Psr\Container\ContainerInterface`

Comment (lucasmichot) [inline src/Illuminate/Contracts/Cache/Repository.php]: Nevertheless @deleugpn make sure to add this `"psr/simple-cache": "~1.0"` in `Illuminate/Contracts/composer.json` too 😉

Comment (deleugpn) [inline src/Illuminate/Contracts/Cache/Repository.php]: Done :)

**Comment by taylorotwell** [comment]
What benefits will this provide in terms of interop right now. Which libraries are using PSR-16?

**Comment by lucasmichot** [comment]
Some 120 packages : https://packagist.org/packages/psr/simple-cache/dependents

**Comment by garygreen** [comment]
Sigh, another PSR that no one is going to care about 😄

**Comment by devcircus** [comment]
I appreciate the efforts. In then environment I work in, these interoperability patches really come in handy.

**Comment by deleugpn** [comment]
I guess I cannot really answer to the question about interop. My proposal came from reading the survey from the PSR (which Laravel was included) and noticed that it was just a few steps away from becoming fully compatible:

https://docs.google.com/spreadsheet/ccc?key=0Ak2JdGialLildEM2UjlOdnA4ekg3R1Bfeng5eGlZc1E#gid=0

**Comment by garygreen** [comment]
> In then environment I work in, these interoperability patches really come in handy.

Examples?

**Pull request merged**